test/basic.test.ts

```diff
@@ -2536,6 +2536,12 @@ describe('component islands', () => {
     await islandPageRequest
     await page.locator('#server-page').waitFor()
   })
+
+  it('should show error on 404 error for server pages during client navigation', async () => {
```

Comment:
currently the 'lost page' renders differently on client-side navigation vs hard reload. (the message is different, for example)

ideally we'd make it the same - do you have any ideas for that?